shared/util.go

```diff
@@ -289,28 +300,43 @@ func ParseLXDFileHeaders(headers http.Header) (uid int64, gid int64, mode int, t
 		}
 	}
 
-	mode = -1
+	mode := -1
 	if mode64 >= 0 {
 		mode = int(mode64 & int64(os.ModePerm))
 	}
 
-	type_ = headers.Get("X-LXD-type")
+	filetype := headers.Get("X-LXD-type")
 	/* backwards compat: before "type" was introduced, we could only
 	 * manipulate files
 	 */
-	if type_ == "" {
-		type_ = "file"
+	if filetype == "" {
+		filetype = "file"
+	}
+
+	if !ValueInSlice(filetype, []string{"file", "symlink", "directory"}) {
+		return nil, fmt.Errorf("Invalid file type: %q", filetype)
 	}
 
-	write = headers.Get("X-LXD-write")
+	write := headers.Get("X-LXD-write")
 	/* backwards compat: before "write" was introduced, we could only
 	 * overwrite files
 	 */
 	if write == "" {
 		write = "overwrite"
 	}
 
-	return uid, gid, mode, type_, write
+	if !ValueInSlice(write, []string{"overwrite", "append"}) {
+		return nil, fmt.Errorf("Invalid file write mode: %q", write)
+	}
+
+	return &LXDFileHeaders{
```

Comment:
We should always initialise structs with their field names so that adding new fields in the future doesn't break existing call sites.

Reply:
Fixed. What a marvelous source of subtle bugs! This feels like something we should lint for. I'll take a quick look if I get a minute.

Reply:
Thanks!